modin/core/io/text/text_file_dispatcher.py

```diff
@@ -997,6 +997,10 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 filepath_or_buffer, callback=cls.read_callback, **kwargs
             )
 
+        # We know for sure that the buffer is not used further in the code,
+        # renaming in order not to be misleading
+        filepath = filepath_or_buffer_md
```

Comment:
if the buffer isn't used, why do we try to extract it a few lines above? (`cls.get_path_or_buffer`)
https://github.com/modin-project/modin/blob/97769988a6f19e4b76f34238c97bf159ee7626a5/modin/core/io/text/text_file_dispatcher.py#L960-L964
We don't use `filepath_or_buffer_md` as a buffer **nowhere** in this function, not only after this renaming line. 

BTW, why don't we use the variable as a buffer? What happens if we pass a buffer? If the case is not supported, maybe we should raise a NonImplementedError on it instead of silently ignoring the fact that `filepath_or_buffer_md` may contain a file buffer?

Reply:
`cls.get_path_or_buffer` tries to extract the filename in order to continue working with the string. However, if this cannot be done, a buffer is returned, which will fall into the check of supported types (into `check_parameters_support` function) and further into pandas. So a buffer **can be used** in the function before `check_parameters_support`+`single_worker_read`.

Buffer data type isn't serializable - root cause.
We do not silently ignore,  this case is executed in pandas.

Reply:
I'm concerned that both of the variable names would be available in the function's namespace pointing to the same object (`filepath` and `filepath_or_buffer_md`), leaving a developer a choice of which one to use. That could end up in even more confusion if both of the variables be present in the flow mixed together (if somebody mistakenly would use the old name).

Maybe it's okay, but I don't really like it that way, let's leave an honor to someone else to put the approval :)